### ISSUE TITLE
DocCommentId using C# type aliases for built-in types (sometimes)

### ISSUE THREAD

**Issue by joelmartinez** [jbevain/cecil#370]
TL;DR … we're [observing](https://github.com/mono/api-doc-tools/issues/51) some instances where the C# type alias is substituted in different sections of the code. Case in point (these values are being written to XML, thus the `&lt;`):

```
M:System.Dynamic.ExpandoObject.System#Collections#Generic#ICollection&lt;System#Collections#Generic#KeyValuePair&lt;string,object&gt;&gt;#Add(System.Collections.Generic.KeyValuePair{System.String,System.Object})

M:System.Dynamic.ExpandoObject.System#Collections#Generic#ICollection&lt;System#Collections#Generic#KeyValuePair&lt;System#String,System#Object&gt;&gt;#Add(System.Collections.Generic.KeyValuePair{System.String,System.Object})
```
Note the subtle differences of `System#String,System#Object` and `string,object` in the two strings above.

I've pored over both my code, and the code of `DocCommentId`, and haven't found the root cause yet … and here's the thing, have been unable to reproduce this in smaller-scale tests 😳

In any case, was really hoping maybe there was some insight you could add, @jbevain, since you originally wrote this class. Is there any case you're aware of, that a `.FullName` for one of the built in types would be the C# version rather than the FQN? or a replacement that might happen anywhere in any circumstances?

**Comment by jbevain** [comment]
Well I don't see anything in DocComentId that uses aliases, so I have no clue :)

Maybe the one with the alias are generated by another tool? In that case, we'd need to know the rules for using aliases.

**Comment by joelmartinez** [comment]
They were both generated by `DocCommentId.GetDocCommendId` ... in two different parts of the code. Very strange, as I haven't found any indications that the value is being modified after this call (it's a big and complicated codebase :P ). 

Ok, well that's fair, thanks for taking a look at it ... I was mainly just trying to see if there was some insight you might have had which could have answered the conundrum. I'll keep trying to find it. Thanks!

**Comment by joelmartinez** [comment]
@jbevain just in case you were curious, wanted to ping you for FYI. I did eventually find the root cause of this issue, and it turns out that mono's compiler is emitting slightly different metadata on a few things. cecil is just (rightly) reporting what is on disk.

https://github.com/mono/api-doc-tools/issues/51#issuecomment-290731863

So to be clear, definitely not a cecil bug ... but wanted to make you aware in case you encounter a similar issue elsewhere :)

**Comment by jbevain** [comment]
Thanks for the heads-up!
